src/app/order/order.component.ts

```diff
@@ -1,29 +1,44 @@
 import { Component, OnInit, OnDestroy } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { FormGroup, FormBuilder, AbstractControl, Validators } from '@angular/forms';
+import {
+  FormBuilder,
+  AbstractControl,
+  Validators,
+  FormGroup,
+  FormControl,
+} from '@angular/forms';
 
 import { RestaurantService } from '../restaurant/restaurant.service';
 import { Restaurant } from '../restaurant/restaurant';
 import { OrderService, Order, Item } from './order.service';
 import { Subject } from 'rxjs';
-import { takeUntil } from "rxjs/operators";
-import { ItemTotalPipe } from "../shared/item-total.pipe";
+import { takeUntil } from 'rxjs/operators';
+import { ItemTotalPipe } from '../shared/item-total.pipe';
 
+interface OrderForm {
+  restaurant: FormControl<string | undefined>;
+  name: FormControl<string>;
+  address: FormControl<string>;
+  phone: FormControl<string>;
+  items: FormControl<Item[]>;
+}
 
-const minLengthArray = (min: number) => (c: AbstractControl): {[key: string]: any} | null => {
-  if (c.value.length >= min) {
-    return null;
-  }
-  return { minLengthArray: {valid: false }};
-};
+const minLengthArray =
+  (min: number) =>
+  (c: AbstractControl): { [key: string]: any } | null => {
+    if (c.value.length >= min) {
+      return null;
+    }
+    return { minLengthArray: { valid: false } };
+  };
 
 @Component({
   selector: 'pmo-order',
   templateUrl: './order.component.html',
-  styleUrls: ['./order.component.less']
+  styleUrls: ['./order.component.less'],
 })
 export class OrderComponent implements OnInit, OnDestroy {
-  orderForm!: FormGroup;
+  orderForm?: FormGroup<OrderForm>;
```

Comment:
seeing how the form gets initialized after an api call resolves, this seems like a very reasonable change, nice